### PR TITLE
fix(predict): Various chart UX fixes / improvements

### PR DESCRIPTION
**Pull request by kevinbluer**
## **Description**

- Various market details chart fixes / improvements 

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes: NA

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

### **After**

https://github.com/user-attachments/assets/665e3b10-522e-4a1b-ad0b-7d6253a99f43

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> Refines PredictDetailsChart visuals and behavior (loading indicator, grid/labels/insets), updates date/percent formatting, and adds comprehensive utils tests.
> 
> - **UI/Chart UX**:
>   - Replace loading placeholder text with `ActivityIndicator`; keep chart skeleton visible during loading.
>   - Tweak layout/spacing: adjust margins (`mb-6`→`mb-4/mb-3`), ensure axis label row min-height, and legend spacing (`mb-3`→`mb-2`).
>   - Update axis label style to `text-[11px]` and use alternative text color.
>   - Refine grid: shift lines (`x1` to `8`), move tick labels up (`y - 4`) and set color to `colors.text.alternative`.
>   - Adjust area gradient fade (`Stop offset` from `100%` to `95%`).
> - **Chart behavior/constants** (`utils.ts`):
>   - Change `LINE_CURVE` alpha to `0.2`; update `CHART_CONTENT_INSET` to `{ top: 8, bottom: 4, left: 8, right: 48 }`.
>   - Revise label formatting: `ONE_WEEK` now `weekday AM/PM`; `MAX` uses 2‑digit year; support seconds/milliseconds.
>   - Add `formatTickValue` precision rules based on range (<1: 2dp, <10: 1dp, ≥10: 0dp).
> - **Tests**:
>   - Update chart tests to check for `line-chart` during loading.
>   - Add comprehensive `utils.test.ts` covering constants, date label formatting, and tick value formatting edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2f1e80afddb689e230ff4013081456c04a262ada. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by matallui — DISMISSED** [review]

File: app/components/UI/Predict/components/PredictDetailsChart/PredictDetailsChart.tsx [hunk 1]
```diff
@@ -125,26 +116,23 @@ const PredictDetailsChart: React.FC<PredictDetailsChartProps> = ({
...
-              <Text
-                key={`placeholder-${index}`}
-                variant={TextVariant.BodyXS}
-                color={TextColor.Alternative}
-              >
-                {label}
-              </Text>
-            ))}
-          </Box>
+            twClassName="px-4 pt-4 pb-0 min-h-[7.75]"
```
[the remark this answers is not in the document]
Comment (kevinbluer) [reply app/components/UI/Predict/components/PredictDetailsChart/PredictDetailsChart.tsx]: 👍

**Review by matallui — APPROVED** [review]
